tests/std/tests/GH_000952_bind_constraints/test.compile.pass.cpp

```diff
@@ -43,3 +54,230 @@ void test() { // COMPILE-ONLY
+static_assert(is_invocable_v<WeirdDual, int, long> == is_invocable_v<decltype(bind_back(WeirdDual{}, 0L)), int>);
+static_assert(
+    is_invocable_v<const WeirdDual, int, long> == is_invocable_v<const decltype(bind_back(WeirdDual{}, 0L)), int>);
+
+// derived from libc++'s test file
+void test_bind_back_deletion() {
+    // Make sure the bind_back unspecified-type is NOT invocable when the call would select a differently-qualified
+    // operator().
+    //
+    // For example, if the call to `operator()() &` is ill-formed, the call to the unspecified-type
+    // should be ill-formed and not fall back to the `operator()() const&` overload.
+    {
+        // Make sure we delete the & overload when the underlying call isn't valid
+        struct F {
+            void operator()() & = delete;
+            void operator()() const& {}
+            void operator()() && {}
+            void operator()() const&& {}
+        };
+        using X = decltype(bind_back(F{}));
+        static_assert(!is_invocable_v<X&>);
+        static_assert(is_invocable_v<X const&>);
+        static_assert(is_invocable_v<X>);
+        static_assert(is_invocable_v<X const>);
+    }
+
+    // There's no way to make sure we delete the const& overload when the underlying call isn't valid,
+    // so we can't check this one.
+
+    // Make sure we delete the && overload when the underlying call isn't valid
+    {
+        struct F {
+            void operator()() & {}
+            void operator()() const& {}
+            void operator()() && = delete;
+            void operator()() const&& {}
+        };
+        using X = decltype(bind_back(F{}));
+        static_assert(is_invocable_v<X&>);
+        static_assert(is_invocable_v<X const&>);
+        static_assert(!is_invocable_v<X>);
+        static_assert(is_invocable_v<X const>);
+    }
+
+    // Make sure we delete the const&& overload when the underlying call isn't valid
+    {
+        struct F {
+            void operator()() & {}
+            void operator()() const& {}
+            void operator()() && {}
+            void operator()() const&& = delete;
+        };
+        using X = decltype(bind_back(F{}));
+        static_assert(is_invocable_v<X&>);
+        static_assert(is_invocable_v<X const&>);
+        static_assert(is_invocable_v<X>);
+        static_assert(!is_invocable_v<X const>);
+    }
+
+    // Some examples by Tim Song
+    {
+        struct T {};
+        struct F {
+            void operator()(T&&) const& {}
+            void operator()(T&&) && = delete;
+        };
+        using X = decltype(bind_back(F{}));
+        static_assert(!is_invocable_v<X, T>);
+    }
+
+    {
+        struct T {};
+        struct F {
+            void operator()(T const&) const {}
+            void operator()(T&&) const = delete;
+        };
+        using X = decltype(bind_back(F{}, T{}));
+        static_assert(!is_invocable_v<X>);
+    }
+}
+#endif // _HAS_CXX23
+#endif // _HAS_CXX20

```

Comment:
Suggestion: The nesting isn't necessary here, and the test would be slightly clearer without it.